src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5391,7 +5400,13 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
                 TypeWithState rightState;
                 if (!node.IsRef)
                 {
-                    rightState = VisitOptionalImplicitConversion(right, leftLValueType, UseLegacyWarnings(left, leftLValueType), trackMembers: true, AssignmentKind.Assignment);
+                    var discarded = left is BoundDiscardExpression;
+                    rightState = VisitOptionalImplicitConversion(right, targetTypeOpt: discarded ? default : leftLValueType, UseLegacyWarnings(left, leftLValueType), trackMembers: true, AssignmentKind.Assignment);
+
+                    if (left is BoundDiscardExpression discard)
+                    {
+                        SetAnalyzedNullability(discard, new VisitResult(rightState), isLvalue: true);
+                    }
```

Comment:
Is this different from the `SetResult` below?

Reply:
One is for the discard `_` and the other is for the assignment expression `_ = expr`.

---
In reply to: [303490371](https://github.com/dotnet/roslyn/pull/37214#discussion_r303490371) [](ancestors = 303490371)